tools/launch.py

```diff
@@ -137,6 +142,12 @@ def main():
             'A user has to specify a partition configuration file with --part_config.'
     assert args.ip_config is not None, \
             'A user has to specify an IP configuration file with --ip_config.'
+    if args.num_omp_threads is None:
+        # Here we assume all machines have the same number of CPU cores as the machine
+        # where the launch script runs.
+        args.num_omp_threads = multiprocessing.cpu_count() // 2 // args.num_trainers
```

Comment:
What will happen when it is 0? (if multiprocessing.cpu_count() // 2 < args.num_trainers.)

Reply:
good point. i'll set a minimum value.